src/utils/message-formatter.js

```diff
@@ -0,0 +1,31 @@
+module.exports = (level = (process.env.DEFAULT_LOG_LEVEL || 'info'), ...args) => {
+  const restArgs = args[0] || '';
+  // if plain text
+  const message = {
+    level,
+    text: restArgs[0],
+    meta: {
+      instanceId: process.env.HOSTNAME
+    }
+  };
+
+  // if error
+  if (restArgs[0] instanceof Error) {
+    Object.assign(message, {
+      text: restArgs[0].message || 'Error: ',
+      meta: {
+        stack: restArgs[0].stack,
+        notify: restArgs[1] ? restArgs[1].notify : true
+      }
+    });
+  }
+  // if more than 1 arg, then all args with index [1+] are included as message meta
+  if (restArgs.length > 1) {
+    const metaArgs = restArgs.slice(1).reduce((sum, next) => Object.assign(sum, next));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This expression might have a value which cannot be called; it might not be a function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.dialonce.io/coding_rules#rule_key=javascript%3AS2873)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this access to "reduce" property, it doesn't exist, as a built-in, on a String. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.dialonce.io/coding_rules#rule_key=javascript%3AS3759)